osu.Game/Screens/Play/HUD/FailingLayer.cs

```diff
@@ -96,18 +102,24 @@ private void updateBindings()
             if (LoadState < LoadState.Ready)
                 return;
 
-            enabled.UnbindBindings();
+            fadePlayfieldWhenHealthLow.UnbindBindings();
 
             // Don't display ever if the ruleset is not using a draining health display.
             if (healthProcessor is DrainingHealthProcessor)
-                enabled.BindTo(configEnabled);
+                fadePlayfieldWhenHealthLow.BindTo(fadePlayfieldWhenHealthLowSetting);
             else
-                enabled.Value = false;
+                fadePlayfieldWhenHealthLow.Value = false;
```

Comment:
Sounds like this is just adding unnecessary complexity to the visibility logic now, why not just add to the check inside `updateState` and call that method on `BindHealthProcessor` as well?

Reply:
I fail to see the gain, honestly. It would amount to unnecessarily conflating ruleset-specific logic with visibility logic in my view.

Reply:
for readability, having both `fadePlayfieldWhenHealthLow` and `fadePlayfieldWhenHealthLowSetting` together really makes thing unnecessarily complex, you can get rid of that and get rid of the whole `updateBindings()` method, and then very simply add `healthProcessor is DrainingHealthProcessor` check to `updateState()` in replace.

Reply:
I thought it would look worse than it actually does. Combining three `bool`s has me a little nervous but it passes tests and seems to work ok, so sure.